server/src-lib/Hasura/RQL/DDL/CustomTypes.hs

```diff
@@ -15,20 +15,26 @@ import qualified Database.PG.Query                 as Q
 import qualified Language.GraphQL.Draft.Syntax     as G
 
 import           Hasura.EncJSON
+import           Hasura.GraphQL.Validate.Types     (mkScalarTy)
 import           Hasura.Prelude
 import           Hasura.RQL.Types
 import           Hasura.SQL.Types
 
 import           Hasura.GraphQL.Schema.CustomTypes (buildCustomTypesSchemaPartial)
 
+-- | Validate the custom types and return any reused Postgres base types (as scalars)
 validateCustomTypeDefinitions
   :: (MonadValidate [CustomTypeValidationError] m)
-  => TableCache -> CustomTypes -> m ()
-validateCustomTypeDefinitions tableCache customTypes = do
+  => TableCache -- ^ The Postgres tables
+  -> CustomTypes -- ^ Custom types subject to validation
```

Comment:
I think the role of these two arguments is self-explanatory given their types alone (in combination with the function name), so I’d drop the comments here (since they’re just basically just restating existing information). The other two comments are great, though, thank you for them!

```suggestion
  => TableCache
  -> CustomTypes
```